L_ZWay2.lua

```diff
@@ -1568,6 +1568,12 @@ local function updateChildren (vDevs)
     local status = state and 2 or -1
     local message = state and state .. " failed" or ''
     dev: status_set (status, message)
+    if state then
+      dev: variable_set (SID.HaDevice, "CommFailure", 1)
+      dev: variable_set (SID.HaDevice, "CommFailureTime", os.time())
+    else
+      dev: variable_set (SID.HaDevice, "CommFailure", 0)
+    end		
```

Comment:
I don't think that this implements what you're intending to do.

Several points to note/check:
* these variables will be set every update – it's deperately inefficient
* are you intending to be able to set a watch on any of these variables?
* are you intending that these updates appear in the log?
* AFAIK, CommFailureTime is cleared when the flag itself is cleared?

Reply:
Yes I am actually setting a watch on the CommFailure to send me phone notifications. That's why the status is not sufficient. It's a good point that they get set at every turn. Maybe need a flag to set them only if they change? It is the same as the "status" attribute though and the flag should apply the same way?

Reply:
> Maybe need a flag to set them only if they change?

Yes, that's exactly the point.  Whilst` status_set()` is very fast,` variable_set()` is relatively slow (variables have complex structures and methods to allow logging, watching, archiving, updating UI status...)   The status flag, though, is very much like an attribute, and trivial to set or clear.